azext_iot/tests/iothub/devices/test_hub_export_import.py

```diff
@@ -0,0 +1,328 @@
+                assert (len(dest_modules) == 1)
+
+            for module in orig_modules:
+                target_module = None
+                for mod in dest_modules:
+                    if(module["moduleId"] == mod["moduleId"]):
+                        target_module = mod
+                        break
+
+                assert target_module
+
+                assert (module["authentication"] == target_module["authentication"])
+
+                module_twin = self.cmd(
+                    f"iot hub module-twin show -m {module['moduleId']} -d {device['deviceId']} -n {self.entity_name} -g {self.entity_rg}"
+                ).get_output_in_json()  
+                target_module_twin = self.cmd(
+                    f"iot hub module-twin show -m {module['moduleId']} -d {device['deviceId']} -n {self.dest_hub} -g {self.dest_hub_rg}"
+                ).get_output_in_json()   
+                
+                assert (len(module_twin["properties"]["desired"]) == len(target_module_twin["properties"]["desired"]))
+                for prop in module_twin["properties"]["desired"]:
+                    if(prop != "$metadata" and prop != "version"):
+                        assert (prop in target_module_twin["properties"]["desired"])
+                        assert (module_twin["properties"]["desired"][prop] == target_module_twin["properties"]["desired"][prop])
+                
+                if("tags" in module_twin.keys()):
+                    assert (module_twin["tags"] == target_module_twin["tags"])
+                assert (module_twin["status"] == target_module_twin["status"])
+
+    def clean_up_dest_hub(self):
+
+        if not settings.env.azext_iot_desthub:
+            self.cmd("iot hub delete --name {} --resource-group {}".format(self.dest_hub, self.entity_rg))
+        else: 
+            dest_hub_configs = self.cmd(
+                f"iot hub configuration list -n {self.dest_hub} -g {self.dest_hub_rg}"
+            ).get_output_in_json()
+
+            for config in dest_hub_configs:
+                self.cmd(
+                    "iot hub configuration delete -c {} -n {} -g {}".format(config["id"], self.dest_hub, self.dest_hub_rg)
+                )
+
+            dest_hub_identities = self.cmd(
+                f"iot hub device-identity list -n {self.dest_hub} -g {self.dest_hub_rg}"
+            ).get_output_in_json()
+            
+            for id in dest_hub_identities:
+                self.cmd(
+                    "iot hub device-identity delete -d {} -n {} -g {}".format(id["deviceId"], self.dest_hub, self.dest_hub_rg)
+                )
+
+    def test_import_export(self):
+
+        self.create_hub_state()
+
+        # self.cmd(
+        #     f"iot hub state export -n {self.entity_name} -f {self.filename} -g {self.entity_rg}"
+        # )
+
+        # self.cmd(
+        #     f"iot hub state import -n {self.dest_hub} -f {self.filename} -g {self.dest_hub_rg} --overwrite"
+        # )
+
+        self.cmd(
```

Comment:
here, you would want to do the 
`for auth_phase in DATAPLANE_AUTH_TYPES:`
so you can check if your command works with different forms of authenticating to the hub

Reply:
Okay, that was my latest question! Is it better stylistically to make those separate tests?

Reply:
stylistically, you would want to have them separate. However, since the export generates the file that you will use in import, it is fine to test both in one test. I would rename the test to reflect that - test_import_export_lifecycle

Reply:
Yeah, but would running it with different auth_phases go in different tests?